test/common/network/udp_fuzz.cc

```diff
@@ -0,0 +1,166 @@
+#include "common/network/udp_packet_writer_handler_impl.h"
+#include "common/network/utility.h"
+
+#include "test/common/network/udp_listener_impl_test_base.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/api/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/threadsafe_singleton_injector.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+//	namespace Network {
+namespace {
+
+class UdpFuzz;
+
+class FuzzUdpListenerCallbacks : public Network::UdpListenerCallbacks {
+public:
+  FuzzUdpListenerCallbacks(UdpFuzz* upf) : my_upf(upf) {}
+  ~FuzzUdpListenerCallbacks() override = default;
+  UdpFuzz* my_upf;
+  void onData(Network::UdpRecvData&& data) override;
+  void onReadReady() override;
+  void onWriteReady(const Network::Socket& socket) override;
+  void onReceiveError(Api::IoError::IoErrorCode error_code) override;
+  void onDataWorker(Network::UdpRecvData&& data) override;
+  void post(Network::UdpRecvData&& data) override;
+  void onDatagramsDropped(uint32_t dropped) override;
+  uint32_t workerIndex() const override;
+  Network::UdpPacketWriter& udpPacketWriter() override;
+};
+
+class UdpFuzz {
+public:
+  UdpFuzz(const uint8_t* buf, size_t len) {
+    // Prepare environment
+    api_ = Api::createApiForTest();
+    dispatcher_ = api_->allocateDispatcher("test_thread");
+    ip_version_ = TestEnvironment::getIpVersionsForTest()[0];
+
+    server_socket_ = createServerSocket(true, ip_version_);
+    server_socket_->addOptions(Network::SocketOptionFactory::buildIpPacketInfoOptions());
+    server_socket_->addOptions(Network::SocketOptionFactory::buildRxQueueOverFlowOptions());
+
+    // Create packet writer
+    udp_packet_writer_ = std::make_unique<Network::UdpDefaultWriter>(server_socket_->ioHandle());
+
+    // Set up callbacks
+    FuzzUdpListenerCallbacks fuzzCallbacks(this);
+
+    // Create listener with default config
+    envoy::config::core::v3::UdpSocketConfig config;
+    std::unique_ptr<Network::UdpListenerImpl> listener_ =
+        std::make_unique<Network::UdpListenerImpl>(dispatcherImpl(), server_socket_, fuzzCallbacks,
+                                                   dispatcherImpl().timeSource(), config);
+
+    Network::Address::Instance* send_to_addr_ = new Network::Address::Ipv4Instance(
+        "127.0.0.1", server_socket_->addressProvider().localAddress()->ip()->port());
+
+    // Now do all of the fuzzing
+    FuzzedDataProvider provider(buf, len);
+    total_packets = provider.ConsumeIntegralInRange<uint16_t>(1, 15);
+    sent_packets = 0;
+    Network::Test::UdpSyncPeer client_(ip_version_);
+    for (uint16_t i = 0; i < total_packets; i++) {
+      std::string packet_ =
+          provider.ConsumeBytesAsString(provider.ConsumeIntegralInRange<uint32_t>(1, 3000));
+      if (packet_.empty()) {
+        packet_ = "EMPTY_PACKET";
```

Comment:
is there any advantage to just capping out at remaining bytes?

Reply:
I don't think so. It shouldn't add to the possible set of inputs, and I think this is the easier solution.